--- source/src/remote/inet.cpp
+++ destination/src/remote/inet.cpp
@@ -2351,7 +2351,7 @@
 
 	if (strcmp(entry1, host_name))
 	{
-#if (defined UNIX) && !(defined NETBSD)
+#if (defined UNIX) && (defined USEUCLIBC)
 		if (entry1[1] == '@')
 		{
 			if (!innetgr(&entry1[2], host_name, 0, 0))
@@ -2380,7 +2380,7 @@
 
 /* if they're in the user group: + they're in, - they're out */
 
-#if (defined UNIX) && !(defined NETBSD)
+#if (defined UNIX) && (defined USEUCLIBC)
 	if (entry2[1] == '@') {
 		if (innetgr(&entry2[2], 0, user_name, 0)) {
 			if (entry2[0] == '+')
