--- a/src/common/classes/rwlock.h
+++ b/src/common/classes/rwlock.h
@@ -196,7 +196,7 @@
 		if (pthread_rwlockattr_init(&attr))
 			system_call_failed::raise("pthread_rwlockattr_init");
 		// Do not worry if target misses support for this option
-		pthread_rwlockattr_setkind_np(&attr, PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP);
+//		pthread_rwlockattr_setkind_np(&attr, PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP);
 		if (pthread_rwlock_init(&lock, NULL))
 			system_call_failed::raise("pthread_rwlock_init");
 		if (pthread_rwlockattr_destroy(&attr))
