--- a/fatresize.c
+++ b/fatresize.c
@@ -326,11 +326,11 @@ partition_warn_busy(PedPartition* part)
 			("Partition %s is being used.  You must unmount it "
 			 "before you modify it with Parted."),
 			path);
-	ped_free(path);
+	free(path);
 	return 0;
     }
 
-    ped_free(path);
+    free(path);
     return 1;
 }
 
@@ -513,8 +513,8 @@ main(int argc, char **argv)
     }
     else if (!ped_unit_parse(def_str, dev, &end, &range_end))
 	return 1;
-    ped_free(old_str);
-    ped_free(def_str);
+    free(old_str);
+    free(def_str);
 
     printd(3, "ped_geometry_duplicate()\n");
     old_geom = ped_geometry_duplicate(&part->geom);
