--- inadyn-1.99.15/libite/Makefile
+++ inadyn-1.99.15/libite/Makefile.foo
@@ -33,7 +33,7 @@
 CC         ?= $(CROSS)gcc
 AR         ?= $(CROSS)ar
 STRIP      ?= $(CROSS)strip
-INSTALL    := install --backup=off
+INSTALL    := install
 STRIPINST  := $(INSTALL) -s --strip-program=$(CROSS)strip -m 0755
 
 CFLAGS     += -fPIC
@@ -69,7 +69,7 @@
 TARGET      = $(STATICLIB) $(SOLIB)
 
 # Default install paths
-prefix     ?= /usr/local
+prefix     ?= /usr
 libdir     ?= $(prefix)/lib
 datadir    ?= $(prefix)/share/doc/$(LIBNAME)
 incdir     ?= $(prefix)/include
