--- a/config
+++ b/config
@@ -16,7 +16,7 @@
 # IPv6 syntax: [address]:port
 # For example: 0.0.0.0:9999, [::]:9999
 # IPv6 adresses are configured to only allow IPv6 connections
-#hosts = 0.0.0.0:5232
+hosts = 0.0.0.0:5232
 
 # Daemon flag
 #daemon = False
@@ -157,13 +157,13 @@
 #          OTHER BACKENDS ARE NOT READY FOR PRODUCTION.
 # -------
 # Value: filesystem | multifilesystem | database | custom
-#type = filesystem
+type = filesystem
 
 # Custom storage handler
 #custom_handler =
 
 # Folder for storing local collections, created if not present
-#filesystem_folder = ~/.config/radicale/collections
+filesystem_folder = /var/lib/radicale
 
 # Database URL for SQLAlchemy
 # dialect+driver://user:password@host/dbname[?key=value..]
