--- a/preserve.c
+++ b/preserve.c
@@ -109,8 +109,8 @@
                         yr = hist[i].year;
                         mth= hist[i].month+1;
                         if (mth>12) { mth=1; yr++; }
-                        memcpy(&hist[0], &hist[1], sizeof(hist[0])*i);
-                        memset(&hist[i], 0, sizeof(struct hist_rec));
+                        // memcpy(&hist[0], &hist[1], sizeof(hist[0])*i);
+                        // memset(&hist[i], 0, sizeof(struct hist_rec));
                         hist[i].year=yr; hist[i].month=mth; n--;
                     }
                   }
@@ -151,7 +151,9 @@
    char    old_fname[MAXKVAL+4];
    struct  stat hist_stat;
    time_t  now;
-   char    timestamp[48];
+   struct  tm *myts;
+   int     yr ;
+   int     mth;
 
    /* generate 'new' filename */
    sprintf(new_fname, "%s.new", hist_fname);
@@ -170,7 +172,7 @@
 
    /* Generate our timestamp */
    now=time(NULL);
-   strftime(timestamp,sizeof(timestamp),"%d/%b/%Y %H:%M:%S",localtime(&now));
+   myts=localtime(&now);
 
    /* Open file for writing */
    hist_fp = fopen(new_fname,"w");
@@ -179,12 +181,20 @@
       if (verbose>1) printf("%s\n",msg_put_hist);
 
       /* write header */
-      fprintf(hist_fp,"# Webalizer V%s-%s History Data - %s (%d month)\n",
-              version, editlvl, timestamp, HISTSIZE);
+      fprintf(hist_fp,"# Webalizer V%s-%s History Data %04d-%02d-%02d %02d:%02d (%d month)\n",
+              version, editlvl, myts->tm_year + 1900, myts->tm_mon + 1, myts->tm_mday, myts->tm_hour, myts->tm_hour, HISTSIZE);
 
+      // write default empty record is not exists
+      yr = myts->tm_year + 1900;
+      mth= myts->tm_mon + 2;
+      if (mth>12) { mth=1; yr++; }
+      if ( hist[i=HISTSIZE-1].month < mth ) {
+          fprintf(hist_fp,"%d %04d 0 0 0 0 0 0 0 0\n", mth, yr );
+      }
+
       for (i=HISTSIZE-1;i>=0;i--)
       {
-         fprintf(hist_fp,"%d %d %llu %llu %llu %.0f %d %d %llu %llu\n",
+         fprintf(hist_fp,"%d %04d %llu %llu %llu %.0f %d %d %llu %llu\n",
                          hist[i].month,
                          hist[i].year,
                          hist[i].hit,
@@ -267,7 +277,7 @@
                {
                   if (verbose)
                      fprintf(stderr,"Warning! %d month gap detected! "   \
-                             "(%d/%d to %d/%d)\n", n, hist[i].month,
+                             "(%02d/%04d to %02d/%04d)\n", n, hist[i].month,
                              hist[i].year, cur_month, cur_year);
                   if (n>11) hist_gap=1;  /* year or more? */
                }
@@ -277,8 +287,8 @@
                   yr = hist[i].year;
                   mth= hist[i].month+1;
                   if (mth>12) { mth=1; yr++; }
-                  memcpy(&hist[0],&hist[1],sizeof(hist[0])*i);
-                  memset(&hist[i], 0, sizeof(struct hist_rec));
+                  // memcpy(&hist[0],&hist[1],sizeof(hist[0])*i);
+                  // memset(&hist[i], 0, sizeof(struct hist_rec));
                   hist[i].year=yr; hist[i].month=mth; n--;
                }
             }
@@ -351,8 +361,8 @@
    /* first, save the easy stuff */
    /* Header record */
    snprintf(buffer,sizeof(buffer),
-     "# Webalizer V%s-%s Incremental Data - %02d/%02d/%04d %02d:%02d:%02d\n",
-      version,editlvl,cur_month,cur_day,cur_year,cur_hour,cur_min,cur_sec);
+     "# Webalizer V%s-%s Incremental Data - %04d-%02d-%02d %02d:%02d:%02d\n",
+      version,editlvl,cur_year,cur_month,cur_day,cur_hour,cur_min,cur_sec);
    if (fputs(buffer,fp)==EOF) return 1;  /* error exit */
 
    /* Current date/time          */
